### PR TITLE
feat(client): add block/direct PacketDialers and direct PacketListener

### PR DESCRIPTION
**Pull request by fortuna**

### CODE REVIEW

**Comment by fortuna** [comment]
Ping

File: client/go/outline/config/module.go [hunk 1]
```diff
@@ -35,17 +35,19 @@ func newTypeParser[T any](fallbackHandler func(context.Context, configyaml.Confi
...
+	if directSD != nil {
+		directWrappedSD = &Dialer[transport.StreamConn]{ConnectionProviderInfo{ConnTypeDirect, ""}, directSD.DialStream}
+	}
 	streamDialers := newTypeParser(func(ctx context.Context, input configyaml.ConfigNode) (*Dialer[transport.StreamConn], error) {
 		switch input.(type) {
 		case nil:
-			// An absent config implicitly means TCP.
-			return tcpConnDialer, nil
+			// An absent config implicitly means direct access.
+			return directWrappedSD, nil
```
Comment (ohnorobo) [inline client/go/outline/config/module.go:50]: the fact that we can modify this without having to update any tests makes me feel like we really need more tests covering the edge cases of what various configurations mean.

File: client/go/outline/config/module.go [hunk 2]
```diff
@@ -54,11 +56,15 @@ func NewDefaultTransportProvider(tcpDialer transport.StreamDialer, udpDialer tra
...
+	if directPD != nil {
+		directWrappedPD = &Dialer[net.Conn]{ConnectionProviderInfo{ConnTypeDirect, ""}, directPD.DialPacket}
+	}
 	packetDialers := newTypeParser(func(ctx context.Context, input configyaml.ConfigNode) (*Dialer[net.Conn], error) {
 		switch input.(type) {
 		case nil:
-			// An absent config implicitly means UDP.
-			return &Dialer[net.Conn]{ConnectionProviderInfo{ConnTypeDirect, ""}, udpDialer.DialPacket}, nil
+			// An absent config implicitly means direct access.
+			return directWrappedPD, nil
```
Comment (ohnorobo) [inline client/go/outline/config/module.go:67]: same here

File: client/go/outline/config/module.go (hunk 1 above)
Comment (fortuna) [reply client/go/outline/config/module.go:50]: This is not really changing the behavior, so tests should not change.

File: client/go/outline/config/module.go (hunk 2 above)
Comment (fortuna) [reply client/go/outline/config/module.go:67]: Ditto.

**Comment by fortuna** [comment]
This doesn't really touch the stuff @jyyi1 is working on. I'd like to merge this sooner (he's away), so we can exercise the limited protection flow.
